src/traces/bar/attributes.js

```diff
@@ -39,6 +39,54 @@ var marker = extendFlat({
         max: 1,
         editType: 'style',
         description: 'Sets the opacity of the bars.'
+    },
+    pattern: {
```

Comment:
The `bar/attributes` is required and used by other traces namely `barpolar`, `box`, `histogram`, `funnel`, `waterfall`.
Are we planning to add `pattern` to any of those?

Reply:
Technically it should be quite easy to apply pattern fill to other plots. I'm planning to support all of them and add some tests!